edge/pkg/devicetwin/dttype/types_helper_test.go

```diff
@@ -0,0 +1,776 @@
+	attrs := make(map[string]*MsgAttr)
+	optional := true
+	metadata := &TypeMetadata{Type: "string"}
+	msgAttr := MsgAttr{Optional: &optional, Metadata: metadata}
+	attrs["SensorTag"] = &msgAttr
+	return attrs
+}
+
+// createDevice is function to create an array of devices.
+func createDevice() []*Device {
+	attrs := createMessageAttribute()
+	devices := []*Device{}
+	device := &Device{ID: "id1", Name: "SensorTag", Description: "Sensor", State: "ON", LastOnline: "TODAY", Attributes: attrs}
+	devices = append(devices, device)
+	return devices
+}
+
+// createMembershipGetResult is function to create MembershipGetResult with given base message.
+func createMembershipGetResult(message BaseMessage) MembershipGetResult {
+	attrs := createMessageAttribute()
+	devices := []*Device{}
+	device := &Device{ID: "id1", Name: "SensorTag", Description: "Sensor", State: "ON", LastOnline: "TODAY", Attributes: attrs}
+	devices = append(devices, device)
+	wantDevice := []Device{}
+	wantDev := Device{ID: device.ID, Name: device.Name, Description: device.Description, State: device.State, LastOnline: device.LastOnline, Attributes: device.Attributes}
+	wantDevice = append(wantDevice, wantDev)
+	memGetResult := MembershipGetResult{BaseMessage: message, Devices: wantDevice}
+	return memGetResult
+}
+
+// TestBuildMembershipGetResult is function to test BuildMembershipGetResult().
+func TestBuildMembershipGetResult(t *testing.T) {
+	baseMessage := BaseMessage{EventID: uuid.NewV4().String(), Timestamp: time.Now().UnixNano() / 1e6}
+	devices := createDevice()
+	memGetResult := createMembershipGetResult(baseMessage)
+	bytesMemGetResult, _ := json.Marshal(memGetResult)
+	tests := []struct {
+		name        string
+		baseMessage BaseMessage
+		devices     []*Device
+		want        []byte
+		wantErr     error
+	}{
+		{
+			name:        "BuildMembershipGetResultTest",
+			baseMessage: baseMessage,
+			devices:     devices,
+			want:        bytesMemGetResult,
+			wantErr:     nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got, err := BuildMembershipGetResult(test.baseMessage, test.devices)
+			if !reflect.DeepEqual(err, test.wantErr) {
+				t.Errorf("BuildMembershipGetResult() error = %v, wantErr %v", err, test.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, test.want) {
+				t.Errorf("BuildMembershipGetResult() = %v, want %v", got, test.want)
+			}
+		})
+	}
+}
+
+//createMessageTwin() is function to create a map of MessageTwin with MetaDataType updated and deleted.
+func createMessageTwin() map[string]*MsgTwin {
+	msgTwins := make(map[string]*MsgTwin)
+	twinMetadataDeleted := MsgTwin{Metadata: &TypeMetadata{Type: "deleted"}}
+	twinMetadataUpdated := MsgTwin{Metadata: &TypeMetadata{Type: "updated"}}
+	msgTwins["empty"] = nil
+	msgTwins["deleted"] = &twinMetadataDeleted
+	msgTwins["updated"] = &twinMetadataUpdated
+	return msgTwins
+}
+
+// createDeviceTwinResultDealType0() is function to create DeviceTwinResult with DealType 0.
```

Comment:
DealType0 -> what is type 0 ? can we have the proper type mentioned here?

Reply:
DealType0 is Get...Updated